worksheet.rb

```diff
@@ -1,34 +1,260 @@
 # Copy your list from above, and in this section
 # determine what data structure each layer should have
 
+# driver -> ID number
+# rides -> date, cost, rating, rider_id
+
 ########################################################
 # Step 3: Make the data structure!
-
 # Setup the entire data structure:
 # based off of the notes you have above, create the
 # and manually write in data presented in rides.csv
 # You should be copying and pasting the literal data
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+ride_data = [
+  {
+    driver_id: "DR0001",
+    rides: [
+      {
+        date: "3rd Feb 2016",
+        cost: 10,
+        rating: 3,
+        rider_id: "RD0003"
+      },
+      {
+        date: "3rd Feb 2016",
+        cost: 30,
+        rating: 4,
+        rider_id: "RD0015"
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 45,
+        rating: 2,
+        rider_id: "RD0003"
+      }
+    ]
+  },
+  {
+    driver_id: "DR0002",
+    rides: [
+      {
+        date: "3rd Feb 2016",
+        cost: 25,
+        rating: 5,
+        rider_id: "RD0073"
+      },
+      {
+        date: "4th Feb 2016",
+        cost: 15,
+        rating: 1,
+        rider_id: "RD0013"
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 35,
+        rating: 3,
+        rider_id: "RD0066"
+      }
+    ]
+  },
+  {
+    driver_id: "DR0003",
+    rides: [
+      {
+        date: "4th Feb 2016",
+        cost: 5,
+        rating: 5,
+        rider_id: "RD0066"
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 50,
+        rating: 2,
+        rider_id: "RD0003"
+      }
+    ]
+  },
+  {
+    driver_id: "DR0004",
+    rides: [
+      {
+        date: "3rd Feb 2016",
+        cost: 5,
+        rating: 5,
+        rider_id: "RD0022"
+      },
+      {
+        date: "4th Feb 2016",
+        cost: 10,
+        rating: 4,
+        rider_id: "RD0022"
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 20,
+        rating: 5,
+        rider_id: "RD0073"
+      }
+    ]
+  }
+]
+
+
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
-
 # Use an iteration blocks to print the following answers:
-# - the number of rides each driver has given
-# - the total amount of money each driver has made
-# - the average rating for each driver
-# - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# QUESTION 1: the number of rides each driver has given
+
+def get_total_rides_per_driver(ride_data)
+  number_of_rides = {}
+  # access driver's IDs to be used as keys in hash "ride_info", then
+  # access the cost of each ride per driver, to be the corresponding value
+  ride_data.length.times do |driver_index|
+    driver_id = ride_data[driver_index][:driver_id]
+    rides = ride_data[driver_index][:rides].length
+    number_of_rides[driver_id] = rides
+  end
```

Comment:
Suggestion:  This works, and it would also be a good chance to use `map` as well to condense the code.